hetzner_invoice/invoice.py

```diff
@@ -27,24 +28,64 @@ def load_transform_invoice(user: str, pw: str, secret: str, filepath: str) -> Da
     logging.info(
         f"Created DataFrame from invoice and server data. Result has {result_df.shape[0]} rows and {result_df.shape[1]} columns. "
     )
-    return result_df
+    result_list = result_df.values.tolist()
+    record_data = []
+    for entry in result_list:
+        record_data.append(tuple(entry))
+    return record_data
 
 
-def save_invoice_to_db(
-    data: pd.DataFrame, user: str, pw: str, host: str, port: str, schema: str
-):
+def save_invoice_to_db(data: List, user: str, pw: str, host: str, schema: str):
     """
     Downloads and transforms data from the most recent Hetzner invoice.
-    :param data: Dataframe with the transformed invoice data
+    :param data: List with the transformed invoice data
     :param user: Username for the database
     :param pw: Password for the database
     :param host: Database host
-    :param port: Database port
     :param schema: Database schema
     """
-    db_url = "mysql+pymysql://{user}:{pw}@{host}:{port}/{schema}".format(
-        user=user, pw=pw, host=host, port=port, schema=schema
-    )
-    logging.info(f"Connecting to {db_url}")
-    engine = create_engine(db_url, echo=False)
-    data.to_sql(name="invoices", con=engine, if_exists="append", index=False)
+    logging.info("Connecting to the database")
+    try:
+        cnx = mysql.connector.connect(
+            user=user, password=pw, host=host, database=schema
+        )
+    except mysql.connector.Error as err:
+        logging.error(f"Error when trying to connect to database: {err}")
+    else:
+        cursor = cnx.cursor()
+        search_duplicate = (
+            "SELECT * FROM `hetzner-invoice`.invoices WHERE type = %s AND description = %s AND id "
```

Comment:
It would be nice to use `invoices` instead of `hetzner-invoice.invoices` because the datatabase name could be different than `hetzner-invoice`